specs/agents/handling-huge-traces/tracing-spans-compress.md

```diff
@@ -87,7 +87,25 @@ boolean isSameKind(Span other) {
 }
 ```
 
-When applying this compression strategy, the `span.name` is set to `Calls to $span.destination.service.resource`.
+When applying this compression strategy, the `span.name` is set to `Calls to <...>` determined from `service.target.*` as follows:
+
+```js
+function getCompositeSpanName(serviceTarget) {
+    const prefix = 'Calls to '
+    if (!serviceTarget.type) {
+        if (!serviceTarget.name) {
+            return prefix + 'unknown'
+        } else {
+            return prefix + serviceTarget.name
+        }
+    } else if (!serviceTarget.name) {
+        return prefix + serviceTarget.type
+    } else {
+        return prefix + serviceTarget.type + '/' + serviceTarget.name
+    }
+}
+```
+
```

Comment:
REVIEW NOTE: This change updates the composite span.name calculation to match the implementation in the Java Agent here: https://github.com/elastic/apm-agent-java/blob/32611fe5174cf7c67f12c885cb6759176749f243/apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/Span.java#L381-L403